### PR TITLE
Updated pull request template and review

### PR DESCRIPTION
**Pull request by lzap**
Adds PR review - moved from (1). Also updates PR template into more
clean and hopefully more readable form. After merging please delete
the page (1) and tell it has been moved here.

https://projects.theforeman.org/projects/foreman/wiki/Reviewing_patches

### CODE REVIEW

File: developer_docs/pr_review.asciidoc [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+- [ ] Scoped_search definitions for virtual fields (:ext_method) use :only_explicit
+- [ ] Makes use of logging appropriately?
+- [ ] Pass block to logger.debug to lazily evaluate expensive debug statements
+- [ ] Deprecations use Foreman::Deprecation with a deadline of latest stable + three versions.
+- [ ] Not changing structure or behavior of the public HTTP API?
+- [ ] No memory or performance concerns?
+- [ ] Any packaging needed?
+- [ ] User documentation needed?
+- [ ] Demo worthy?
+- [ ] Any other actions need to be taken after merge?
```
Comment (ares) [inline developer_docs/pr_review.asciidoc]: I'd suggest adding
- [ ] needs to be added to API?
- [ ] needs hammer support?
- [ ] needs manual update?
- [ ] needs to be in release notes?

File: developer_docs/pr_review.asciidoc [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+- [ ] Redmine opened and in correct project?
+- [ ] No use of .to_sym, .send or other reflection on untrusted inputs.
+- [ ] Not missing string extractions (use :mark_translated: true).
+- [ ] All string extractions follows our rules.
+- [ ] Added appropriate permissions for non-admin users?
+- [ ] New code hasn't been copy-pasted from elsewhere.
+- [ ] Do new fields have appropriate validators?
+- [ ] No exceptions are swaloved/squashed.
+- [ ] Are raised exceptions type of Foreman::Exception or WrappedException?
+- [ ] Unit/functional/integration test(s) present?
```
Comment (ezr-ondrej) [inline developer_docs/pr_review.asciidoc]:
```suggestion
- [ ] Covered with meaningful unit/functional/integration test(s)
```

**Review by ezr-ondrej — COMMENTED** [review]
Thanks for sharing your list with us! :+1: I believe it is very helpful (at least to me now :stuck_out_tongue_closed_eyes: )
1. I would translate the questions to a narrative language. As checking some of the questions feels bit weird.
I am unsure what checking `Any packaging needed?` means? I feel if I check, it needs some, but I would rather have it stating `Necessary packaging done`.

2. Maybe use of nested checkboxing would be nice idea?
Not sure how to group correctly, but something like:
    - Basics
    - Backend
    - API
    - UI/UX
    - Testing
    - After merge

That way we would know, we can check and skip all of the UI category if the PR is not UI related, do not bother with after merge too much until it is ready to merge.

What do you think? :slightly_smiling_face:

File: developer_docs/pr_review.asciidoc (hunk 1 above)
Comment (lzap) [reply developer_docs/pr_review.asciidoc]: Done.

**Comment by lzap** [comment]
All done, rebased.

**Review by ezr-ondrej — DISMISSED** [review]
LGTM :heart:
I can hardly make more notes on the list now. I need to use it for a while.
I wouln't mind merging this, try how it works and lets improve as we go?

Or shall we wait if someone with more experience adds some insight?

File: developer_docs/pr_review.asciidoc [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+- [ ] Fixes the problem described in the issue.
+- [ ] No unrelated changes present.
+- [ ] Redmine opened for the correct project.
+- [ ] No use of .to_sym, .send or other reflection on untrusted inputs.
+- [ ] Not missing string extractions (use :mark_translated: true).
+- [ ] All string extractions follows our rules.
+- [ ] Appropriate permissions for non-admin users added.
+- [ ] New code hasn't been copy-pasted from elsewhere.
+- [ ] All new AR fields have appropriate validators.
+- [ ] No exceptions are swaloved/squashed.
```
Comment (ekohl) [inline developer_docs/pr_review.asciidoc]:
```suggestion
- [ ] No exceptions are swallowed/squashed.
```

File: developer_docs/pr_review.asciidoc [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+[[List]]
+== Review list
+
+- [ ] Ticket number, title and description is correct.
+- [ ] Fixes the problem described in the issue.
+- [ ] No unrelated changes present.
+- [ ] Redmine opened for the correct project.
+- [ ] No use of .to_sym, .send or other reflection on untrusted inputs.
+- [ ] Not missing string extractions (use :mark_translated: true).
```
Comment (ekohl) [inline developer_docs/pr_review.asciidoc]: Should this be in the positive?
```suggestion
- [ ] All strings extracted (use :mark_translated: true).
```

File: developer_docs/pr_review.asciidoc [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+- [ ] All string extractions follows our rules.
```
Comment (ekohl) [inline developer_docs/pr_review.asciidoc]:
```suggestion
- [ ] All string extractions follow our rules.
```

**Comment by lzap** [comment]
Incorporated @ekohl 's fixes, thanks.

**Comment by ezr-ondrej** [comment]
Lets use it for a while and feedback it.
Merging, thanks @lzap :+1:

**Review by ezr-ondrej — APPROVED** [review]

**Pull request merged**